server/src/main/java/org/opensearch/search/approximate/ApproximatePointRangeQuery.java

```diff
@@ -295,40 +295,42 @@ public void intersectRight(PointValues.IntersectVisitor visitor, PointValues.Poi
-                switch (r) {
-                    case CELL_INSIDE_QUERY:
-                    case CELL_CROSSES_QUERY:
-                        if (pointTree.moveToChild() && docCount[0] < size) {
-                            PointValues.PointTree leftChild = pointTree.clone();
-                            // BKD is binary today, so one moveToSibling() is enough to land on the right child.
-                            // If PointTree ever becomes n-ary, update the traversal below to visit all siblings or re-enable a full loop.
-                            if (pointTree.moveToSibling()) {
-                                // We have two children - visit right first
-                                intersectRight(visitor, pointTree, docCount);
-                                // Then visit left if we still need more docs
-                                if (docCount[0] < size) {
-                                    intersectRight(visitor, leftChild, docCount);
-                                }
-                            } else {
-                                // Only one child - visit it
-                                intersectRight(visitor, leftChild, docCount);
-                            }
-                            pointTree.moveToParent();
-                        } else {
-                            if (docCount[0] < size) {
-                                if (r == PointValues.Relation.CELL_INSIDE_QUERY) {
-                                    pointTree.visitDocIDs(visitor);
-                                } else {
-                                    pointTree.visitDocValues(visitor);
-                                }
-                            }
-                        }
-                        break;
-                    case CELL_OUTSIDE_QUERY:
-                        break;
-                    default:
-                        throw new IllegalArgumentException("Unreachable code");
+                if (r == PointValues.Relation.CELL_OUTSIDE_QUERY) {
+                    return;
+                }
+                // Handle leaf nodes
+                if (pointTree.moveToChild() == false) {
+                    if (r == PointValues.Relation.CELL_INSIDE_QUERY) {
+                        pointTree.visitDocIDs(visitor);
+                    } else {
+                        // CELL_CROSSES_QUERY
+                        pointTree.visitDocValues(visitor);
+                    }
+                    return;
+                }
+                // Internal node - get left child reference (we're at left child initially)
+                PointValues.PointTree leftChild = pointTree.clone();
+                // Move to right child if it exists
+                boolean hasRightChild = pointTree.moveToSibling();
+                // For CELL_INSIDE_QUERY, check if we can skip left child
+                if (r == PointValues.Relation.CELL_INSIDE_QUERY && hasRightChild) {
```

Comment:
This section(line 316-325) seems to be meaningless. The code in `intersectLeft` can avoid a `moveToSibling`, but it will not reduce `moveToSibling` in `intersectRight`.